### DISCUSSION TITLE
Performance for complex form assistance required

### DISCUSSION THREAD

**Discussion by BassWow** [react-hook-form/react-hook-form#9019] [Q&A]
Hi guys,

Thank you for such an amazing library, hands down best form library I have worked with to date!

For the sake of brevity I have re-created a project I am working on and taken out a lot of the bells and whistles as I wish to get help on the performance of a rather large form in terms of re-renders and perhaps a better pattern on writing these types of forms.

I have created a codesandbox link below which better illustrates what I am essentially trying to achieve. I just wanted to know what is the best practice for creating FieldArray components as well as main form fields so that on editing certain field the entire form keeps getting re-rendered unless that is expected behaviour? 

I am also using Chakraui for most of my components however I have removed from this example. 

Any assistance is welcomed and thank you so much!

CSB:
[https://codesandbox.io/s/react-hook-form-test-nxpd38](url)

**Comment by Moshyfawn** [comment]
Hey! Thx for the kind words!

While I'm at work, take a look at #7611. I'll try to look at your use-case later today

**Reply by BassWow** [reply]
Thank you and look forward to any insights you can provide :)

**Reply by BassWow** [reply]
@Moshyfawn any luck with this one ? :)
